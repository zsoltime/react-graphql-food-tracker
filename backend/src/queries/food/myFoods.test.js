'use strict';

const myFoods = require('./myFoods');

describe('myFoods query', () => {
  it('returns mathcing foods if filter is provided', async () => {
    const args = { filter: 'peanut' };
    const mockedContext = {
      db: {
        query: {
          foods: jest
            .fn()
            .mockImplementation(() => Promise.resolve([{ name: 'Salted peanuts' }])),
        },
      },
      request: { userId: 'currentUserId' },
    };

    const foods = await myFoods(null, args, mockedContext);

    expect(mockedContext.db.query.foods).toHaveBeenCalled();
    expect(
      mockedContext.db.query.foods.mock.calls[0]
    ).toMatchSnapshot();
    expect(foods).toMatchSnapshot();
  });

  it('throws error if user is not signed in', () => {
    const mockedContext = {
      request: {},
    };

    expect(
      myFoods(null, null, mockedContext)
    ).rejects.toThrowErrorMatchingSnapshot();
  });

  it('returns all foods added by current user', async () => {
    const args = {};
    const mockedContext = {
      db: {
        query: {
          foods: jest
            .fn()
            .mockImplementation(() => Promise.resolve([
              { name: 'FakeFood I' },
              { name: 'FakeFood II' },
            ])),
        },
      },
      request: { userId: 'currentUserId' },
    };

    const foods = await myFoods(null, args, mockedContext);

    expect(mockedContext.db.query.foods).toHaveBeenCalled();
    expect(
      mockedContext.db.query.foods.mock.calls[0]
    ).toMatchSnapshot();
    expect(foods).toMatchSnapshot();
  });
});
