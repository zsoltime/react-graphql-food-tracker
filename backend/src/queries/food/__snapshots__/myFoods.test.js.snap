// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`myFoods query returns all foods added by current user 1`] = `
Array [
  Object {
    "first": 10,
    "orderBy": "createdAt_DESC",
    "skip": undefined,
    "where": Object {
      "user": Object {
        "id": "currentUserId",
      },
    },
  },
]
`;

exports[`myFoods query returns all foods added by current user 2`] = `
Array [
  Object {
    "name": "FakeFood I",
  },
  Object {
    "name": "FakeFood II",
  },
]
`;

exports[`myFoods query returns mathcing foods if filter is provided 1`] = `
Array [
  Object {
    "first": 10,
    "orderBy": "createdAt_DESC",
    "skip": undefined,
    "where": Object {
      "AND": Array [
        Object {
          "OR": Array [
            Object {
              "brand_contains": "peanut",
            },
            Object {
              "name_contains": "peanut",
            },
          ],
        },
        Object {
          "user": Object {
            "id": "currentUserId",
          },
        },
      ],
    },
  },
]
`;

exports[`myFoods query returns mathcing foods if filter is provided 2`] = `
Array [
  Object {
    "name": "Salted peanuts",
  },
]
`;

exports[`myFoods query throws error if user is not signed in 1`] = `"You must be signed in to list foods"`;
